vsintegration/src/FSharp.Editor/CodeFix/RemoveReturnOrYield.fs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading.Tasks
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+
+open FSharp.Compiler.Range
+open FSharp.Compiler.SyntaxTree
+open FSharp.Compiler.SourceCodeServices
+
+[<AutoOpen>]
+module Traversal =
+    type FSharpParseFileResults with
+        member scope.TryRangeOfExprInYieldOrReturn pos =
+            match scope.ParseTree with
+            | Some parseTree ->
+                AstTraversal.Traverse(pos, parseTree, { new AstTraversal.AstVisitorBase<_>() with 
+                    member __.VisitExpr(_path, _, defaultTraverse, expr) =
+                        match expr with
+                        | SynExpr.YieldOrReturn(_, expr, range)
+                        | SynExpr.YieldOrReturnFrom(_, expr, range) when rangeContainsPos range pos ->
+                            Some expr.Range
+                        | _ -> defaultTraverse expr })
+            | None -> None
```

Comment:
This can be moved into thecompiler service and with tests, I just wanted to localize it for easreof development